docker/client.py

```diff
@@ -140,9 +140,14 @@ def build(self, path=None, tag=None, quiet=False, fileobj=None,
             if self._auth_configs:
                 if headers is None:
                     headers = {}
-                headers['X-Registry-Config'] = auth.encode_full_header(
-                    self._auth_configs
-                )
+                if utils.compare_version('1.19', self._version) >= 0:
```

Comment:
Looks like the change went out in Docker 1.7.1, according to the milestone on this PR: https://github.com/docker/docker/pull/14139

I don't see any switching being done on the API version, so it's possible that backwards compatibility is broken. I've asked for clarification: https://github.com/docker/docker/pull/14139#issuecomment-123246739